src/mongo/delegates/SearchIndexer.class.php

```diff
@@ -0,0 +1,293 @@
+<?php
+
+namespace Tripod\Mongo\Composites;
+
+require_once TRIPOD_DIR . 'mongo/MongoTripodConstants.php';
+require_once TRIPOD_DIR . 'mongo/delegates/SearchDocuments.class.php';
+require_once TRIPOD_DIR . 'mongo/providers/MongoSearchProvider.class.php';
+require_once TRIPOD_DIR . 'exceptions/SearchException.class.php';
+
+use Tripod\Mongo\Config;
+use Tripod\Mongo\ImpactedSubject;
+use Tripod\Mongo\Labeller;
+/**
+ * Class SearchIndexer
+ * @package Tripod\Mongo\Composites
+ */
+class SearchIndexer extends CompositeBase
+{
+    private $tripod = null;
+
+    protected $labeller = null;
+
+    protected $stat = null;
+
+    /**
+     * @var $configuredProvider \Tripod\ISearchProvider
+     */
+    private $configuredProvider = null;
+
+    /**
+     * @param \Tripod\Mongo\Driver $tripod
+     * @throws \Tripod\Exceptions\SearchException
+     */
+    public function __construct(\Tripod\Mongo\Driver $tripod)
+    {
+        $this->tripod = $tripod;
+        $this->storeName = $tripod->getStoreName();
+        $this->podName = $tripod->podName;
+        $this->labeller = new Labeller();
+        $this->stat = $tripod->getStat();
+        $this->config = Config::getInstance();
+        $provider = $this->config->getSearchProviderClassName($this->tripod->getStoreName());
```

Comment:
We've had issues with these dynamic function names previously. Not in their use, but in that you can't search for the function name in your IDE. Is there any documentation we can add here so it'll be searchable?